### PR TITLE
Fixing BWC Test suite runner

### LINKED ISSUE #536: [BUG] BWC test suite does not run on Jenkins
**Describe the bug**
BWC test job fails to run Jenkins due to misconfigured directory to kick off tests. 

```
Running ./bundle-workflow/src/run_bwc_test.py --s3-bucket <artifact_s3_bucket> --opensearch-version 1.1.0 --build-id 163 --architecture x64 --test-run-id 6 ...
2021-09-17 21:50:08 INFO     Switching to temporary work_dir: /tmp/tmp70l5b7_e
Traceback (most recent call last):
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/run_bwc_test.py", line 31, in <module>
    sys.exit(main())
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/run_bwc_test.py", line 26, in main
    args.s3_bucket, args.build_id, args.opensearch_version, args.architecture, cur_dir)
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/manifests/bundle_manifest.py", line 85, in from_s3
    S3Bucket(bucket_name).download_file(manifest_s3_path, work_dir)
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/aws/s3_bucket.py", line 99, in download_file
    local_dir = Path(dest)
  File "/usr/lib64/python3.7/pathlib.py", line 1027, in __new__
    self = cls._from_parts(args, init=False)
  File "/usr/lib64/python3.7/pathlib.py", line 674, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "/usr/lib64/python3.7/pathlib.py", line 658, in _parse_args
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**To Reproduce**
Run `/bundle-workflow/src/test.sh bwc-test --s3-bucket <artifact_s3_bucket> --opensearch-version 1.1.0 --build-id 163 --architecture x64 --test-run-id 6`

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
- BWC test suite does not run on Jenkins due to mis-configured directory.
- Also fixes the path to read downloaded manifest predictable.

Before:
```
Running ./bundle-workflow/src/run_bwc_test.py --s3-bucket <artifact_s3_bucket> --opensearch-version 1.1.0 --build-id 163 --architecture x64 --test-run-id 6 ...
2021-09-17 21:50:08 INFO     Switching to temporary work_dir: /tmp/tmp70l5b7_e
Traceback (most recent call last):
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/run_bwc_test.py", line 31, in <module>
    sys.exit(main())
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/run_bwc_test.py", line 26, in main
    args.s3_bucket, args.build_id, args.opensearch_version, args.architecture, cur_dir)
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/manifests/bundle_manifest.py", line 85, in from_s3
    S3Bucket(bucket_name).download_file(manifest_s3_path, work_dir)
  File "/var/jenkins/workspace/bwc-test/bundle-workflow/src/aws/s3_bucket.py", line 99, in download_file
    local_dir = Path(dest)
  File "/usr/lib64/python3.7/pathlib.py", line 1027, in __new__
    self = cls._from_parts(args, init=False)
  File "/usr/lib64/python3.7/pathlib.py", line 674, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "/usr/lib64/python3.7/pathlib.py", line 658, in _parse_args
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

After:
```
Running ./bundle-workflow/src/run_bwc_test.py --s3-bucket <artifact_s3_bucket> --opensearch-version 1.1.0 --build-id 163 --architecture x64 --test-run-id 6 ...
2021-09-20 23:33:43 INFO     Switching to temporary work_dir: /tmp/tmpt8ckozxc
2021-09-20 23:33:44 INFO     Executing "git init" in /tmp/tmpt8ckozxc/OpenSearch
2021-09-20 23:33:44 INFO     Executing "git remote add origin https://github.com/opensearch-project/OpenSearch.git" in /tmp/tmpt8ckozxc/OpenSearch
2021-09-20 23:33:44 INFO     Executing "git fetch --depth 1 origin c10346550558ec24bba9b0ed36dbe880a24756a1" in /tmp/tmpt8ckozxc/OpenSearch
2021-09-20 23:33:46 INFO     Executing "git checkout FETCH_HEAD" in /tmp/tmpt8ckozxc/OpenSearch
2021-09-20 23:33:47 INFO     Executing "git rev-parse HEAD" in /tmp/tmpt8ckozxc/OpenSearch
2021-09-20 23:33:47 INFO     Checked out https://github.com/opensearch-project/OpenSearch.git@c10346550558ec24bba9b0ed36dbe880a24756a1 into /tmp/tmpt8ckozxc/OpenSearch at c10346550558ec24bba9b0ed36dbe880a24756a1
```
 
### Issues Resolved
Closes #536 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/src/manifests/bundle_manifest.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def from_s3(bucket_name, build_id, opensearch_version, architecture, work_dir=No
             build_id, opensearch_version, architecture
         )
         S3Bucket(bucket_name).download_file(manifest_s3_path, work_dir)
-        with open("manifest.yml", "r") as file:
+        with open(work_dir + "/manifest.yml", "r") as file:
```
Comment (VachaShah) [inline bundle-workflow/src/manifests/bundle_manifest.py:86]: Will this cause any changes for integ test?

Comment (dblock) [inline bundle-workflow/src/manifests/bundle_manifest.py:86]: Should be `os.path.join()`

Actually change this to be `BundleManifest.from_path(os.path.join(work_dir, 'manifest.yml'))`

**Review by dblock — CHANGES_REQUESTED** [review]

File: bundle-workflow/src/manifests/bundle_manifest.py (hunk 1 above)
Comment (dblock) [reply bundle-workflow/src/manifests/bundle_manifest.py:86]: One more thing. This whole `download_file` business should be 

1. Download to a _temporary directory/file name_.
2. Load via `from_path`.

This way it doesn't pollute current directory from where it's called.

**Review by dblock — APPROVED** [review]

File: bundle-workflow/src/manifests/bundle_manifest.py (hunk 1 above)
Comment (saratvemulapalli) [reply bundle-workflow/src/manifests/bundle_manifest.py]: Will create a follow issue for this.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**